File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    
    const dfs =(now, depth) => {
        if (depth == numbers.length){
            if (now == target) answer++;
            return
        }
        dfs(now+numbers[depth], depth+1)
        dfs(now-numbers[depth], depth+1)
    }
    
    
    var answer = 0;
    dfs(0,0)

    return answer;
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    global answer
    answer = 0
    def dfs(sum, depth):
        global answer
        if depth == len(numbers):
            if sum == target: 
                answer+=1
            return
        

        dfs(sum+numbers[depth], depth+1)
        dfs(sum-numbers[depth], depth+1)
            
    dfs(0,0)
    return answer